app/src/main/java/org/schabi/newpipe/util/ShareUtils.java

```diff
@@ -208,24 +241,45 @@ private static String getDefaultAppPackageName(final Context context, final Inte
     /**
      * Open the android share menu to share the current url.
      *
-     * @param context the context to use
-     * @param subject the url subject, typically the title
-     * @param url     the url to share
+     * @param context         the context to use
+     * @param subject         the url subject, typically the title
+     * @param url             the url to share
+     * @param imagePreviewUrl the image of the subject
      */
-    public static void shareText(final Context context, final String subject, final String url) {
-        shareText(context, subject, url, true);
+    public static void shareText(final Context context,
+                                 final String subject,
+                                 final String url,
+                                 final String imagePreviewUrl) {
+        shareText(context, subject, url, imagePreviewUrl, true);
     }
 
-
+    /**
+     * Open the android share sheet to share the current url.
+     *
+     * For Android 10+ users, a content preview is shown, which includes the title of the shared
+     * content.
+     * Support sharing the image of the content needs to done, if possible.
+     *
+     * @param context         the context to use
+     * @param subject         the url subject, typically the title
+     * @param url             the url to share
+     * @param imagePreviewUrl the image of the subject
+     * @param showPreviewText show the subject as an extra title of the Android share sheet if true
+     */
     public static void shareText(final Context context,
                                  final String subject,
                                  final String url,
+                                 final String imagePreviewUrl,
                                  final boolean showPreviewText) {
         final Intent shareIntent = new Intent(Intent.ACTION_SEND);
         shareIntent.setType("text/plain");
-        if (!subject.isEmpty() && showPreviewText) {
-            shareIntent.putExtra(Intent.EXTRA_SUBJECT, subject);
+        if (!imagePreviewUrl.isEmpty() && !subject.isEmpty() && showPreviewText) {
             shareIntent.putExtra(Intent.EXTRA_TITLE, subject);
+            /* TODO: add the image of the content to Android share sheet with setClipData after
+                generating a content URI of this image, then use ClipData.newUri(the content
+                resolver, null, the content URI) and set the ClipData to the share intent with
+                shareIntent.setClipData(generated ClipData).*/
+            //shareIntent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
```

Comment:
Does someone have an idea how to do this, and do not download again the image of the content if possible?

Reply:
@TiA4f8R I don't know if this is doable, it would require the library we are using for images to provide ContentUris of cached images. I'd skip this for now and create a different PR later.

Reply:
Ok, I will let you do this later.

Reply:
> Ok, I will let you do this later. 

Ok, I will try, though I don't promise anything ;-)

> Should I keep the comment?

Yes, keep the comment, thank you